### PR TITLE
node: updates for node.Config

### PR DESCRIPTION
**Pull request by Wondertan**
## Context
Work on #30 turns out quite big and to follow the practice of keeping PR relatively small, I've extracted some work related to configs into separate PR

## Changes
* Simplify and prettify FullNode tests 
* Extract node.Config into a new subpackage to avoid circular dependencies
* De/serialization functions for Config 
* Use Config value everywhere instead of pointers
	* Pointers create issues with serialization which can be fixed using [mapstructure](https://github.com/mitchellh/mapstructure), but that is some additional complexity that does no make sense for me.
	* If you have arguments against using values for Config - pls bring them up, because I don't get the.
* New FX utility

### CODE REVIEW

File: node/full.go [hunk 1]
```diff
@@ -1,39 +1,32 @@
...
+	"github.com/celestiaorg/celestia-node/node/config"
+	nodecore "github.com/celestiaorg/celestia-node/node/core"
+	"github.com/celestiaorg/celestia-node/node/fxutil"
 	"github.com/celestiaorg/celestia-node/service/block"
 )
 
 // NewFull assembles a new Full Node from required components.
-func NewFull(cfg *Config) (*Node, error) {
-	return newNode(fullComponents(cfg))
+func NewFull(cfg *config.Config, corecfg *core.Config) (*Node, error) {
```
Comment (renaynay) [inline node/full.go:14]: Agh i don't really like this cause we have a goal to also run celestia full node standalone (without a trusted core process) - can `core config` be part of the overall `Config`?

**Review by renaynay — APPROVED** [review]
I'm okay with these changes. I don't really like passing the core config directly to the new full node but it's fine.

**Review by liamsi — APPROVED** [review]
👍🏼

**Comment by Wondertan** [comment]
I've moved back the Config to node, as there is another solution to dep cycle and I still prefer to keep Config in the node package, so it is clear that is related to the node by just looking into `node.Config`.

**Comment by Wondertan** [comment]
Also, I've changed the toml parser to BurntSushi/toml from pelletier/go-toml/v2 as the latter does not support Config parsing of Tendermint config needed in https://github.com/celestiaorg/celestia-node/pull/81, as having two parsers is not good, even if the latter is much better in terms of performance.

**Review by renaynay — APPROVED** [review]

File: node/config.go [hunk 2]
```diff
@@ -19,3 +23,18 @@ func DefaultConfig() *Config {
 		Core: core.DefaultConfig(),
 	}
 }
+
+// TODO(@Wondertan): We should have a description for each field written into w,
+//  so users can instantly understand purpose of each field. Ideally, we should have a utility program to parse comments
+//  from actual sources(*.go files) and generate docs from comments. Hint: use 'ast' package.
```
Comment (liamsi) [inline node/config.go:29]: I'm pretty sure something like this already exists.

**Review by liamsi — APPROVED** [review]
👏🏼 nothing to complain about.

**Comment by Wondertan** [comment]
Shipit-tipihS

**Pull request merged**